core/src/main/scala/kafka/server/QuotaFactory.scala

```diff
@@ -52,7 +58,8 @@ object QuotaFactory extends Logging {
       new ClientQuotaManager(clientProduceConfig(cfg), metrics, Produce, time),
       new ClientRequestQuotaManager(clientRequestConfig(cfg), metrics, time),
       new ReplicationQuotaManager(replicationConfig(cfg), metrics, LeaderReplication, time),
-      new ReplicationQuotaManager(replicationConfig(cfg), metrics, FollowerReplication, time)
+      new ReplicationQuotaManager(replicationConfig(cfg), metrics, FollowerReplication, time),
+      new ReplicationQuotaManager(replicationConfig(cfg), metrics, IntraBrokerReplication, time)
```

Comment:
I don't have a good reason for adding a separate window size and window samples. I think the main difference between interBroker and logDirsBalancing quota is the throttle rate which can already be configured differently for these two quotas. Is there any scenario where user may want to use different window size and window samples for interBroker and logDirsBalancing quota? If not, maybe we can do it when we need it in the future?

Reply:
The window size and window samples impact how much load can be put during initialization. During initialization, we give a full window worth of quota. So, the larger the window, the more bytes can be put in initially. Since interBroker quota may be configured based on network capacity whereas logDirsBalancing quota is mostly based on disk capacity, having separate window sizes between the two quotas allows the admin to control the initial load separately.

Reply:
@junrao I see. The default size of the full window is 11 seconds. I think the replica movement typically lasts much longer than 11 seconds and in general it is not a big deal to have a slow start. I am not sure it is worth two additional configs to optimize the initial performance of intra-broker replica movement.

I don't have a strong opinion on this. Do you prefer me to add two configs for this new quota?

Reply:
Yes, I'd prefer to add two new configs for the new quota since we already have separate window configs for the client and the replication quota.

Reply:
Sure. I will update the patch to add these two new configs.